addons/website_sale/models/product.py

```diff
@@ -441,11 +441,13 @@ def _compute_product_website_url(self):
     @api.multi
     def website_publish_button(self):
         self.ensure_one()
-        res = self.product_tmpl_id.website_publish_button()
+        return self.product_tmpl_id.website_publish_button()
 
-        # res can be the result of write() if not website_publisher
-        if type(res) == dict and res.get('type') == 'ir.actions.act_url':
-            res['url'] = self.website_url
+    @api.multi
+    def open_website_url(self):
+        self.ensure_one()
+        res = self.product_tmpl_id.open_website_url()
+        res['url'] = self.website_url
```

Comment:
Is this needed ? Why was not it there before ?

Reply:
It is not needed right now but if someone uses open_website_url on a product.product in the future it will lead him to the right one and not the template.